lib/ansible/modules/cloud/amazon/ecs_taskdefinition_facts.py

```diff
@@ -0,0 +1,346 @@
+networkMode:
+    description: Network mode for the containers
+    returned: always
+    type: string
+revision:
+    description: Revision number that was queried
+    returned: always
+    type: int
+volumes:
+    description: The list of volumes in a task
+    returned: always
+    type: complex
+    contains:
+        name:
+            description: The name of the volume.
+            returned: when present
+            type: string
+        host:
+            description: The contents of the host parameter determine whether your data volume persists on the host container instance and where it is stored.
+            returned: when present
+            type: bool
+        sourcePath:
+            description: The path on the host container instance that is presented to the container.
+            returned: when present
+            type: string
+status:
+    description: The status of the task definition
+    returned: always
+    type: string
+requiresAttributes:
+    description: The container instance attributes required by your task
+    returned: when present
+    type: complex
+    contains:
+        name:
+            description: The name of the attribute.
+            returned: when present
+            type: string
+        value:
+            description: The value of the attribute.
+            returned: when present
+            type: string
+        targetType:
+            description: The type of the target with which to attach the attribute.
+            returned: when present
+            type: string
+        targetId:
+            description: The ID of the target.
+            returned: when present
+            type: string
+placementConstraints:
+    description: A list of placement constraint objects to use for tasks
+    returned: always
+    type: complex
+    contains:
+        type:
+            description: The type of constraint.
+            returned: when present
+            type: string
+        expression:
+            description: A cluster query language expression to apply to the constraint.
+            returned: when present
+            type: string
+'''  # NOQA
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+class EcsTaskManager:
+    """Handles ECS Tasks"""
+
+    def __init__(self, module):
+        self.module = module
+
+        try:
+            region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+            if not region:
+                module.fail_json(msg="Region must be specified as a parameter, in EC2_REGION or AWS_REGION environment variables or in boto configuration file")
+            self.ecs = boto3_conn(module, conn_type='client', resource='ecs', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+        except boto3.exception.NoAuthHandlerFound as e:
```

Comment:
This doesn't actually exist. And won't ever be thrown.

The only error you'll get here is botocore.exceptions.ProfileNotFound

Reply:
Changed the exception caught. The region check seems to be necessary (it failed non-gracefully without it).